tasks/main.yml

```diff
@@ -29,6 +44,58 @@
 - name: Flush handlers
   meta: flush_handlers
 
-- name: Perform join operation
-  shell: echo {{ ad_integration_password }} | realm join {{ ad_integration_realm }}
-  when: not ansible_check_mode
+## Execute other roles if applicable
+- name: Add AD server to existing network connection for DNS
+  include_role:
+   name: fedora.linux_system_roles.network
+  vars:
+    network_connections:
+      - name: "{{ ad_integration_dns_connection_name }}"
+        interface_name: ""
+        type: "{{ ad_integration_dns_connection_type }}"
+        ip:
+          dns: "{{ ad_integration_dns_server }}"
+    network_allow_restart: yes
+  when: ad_integration_manage_dns | bool
+
+- name: Manage timesync
+  include_role:
+   name: fedora.linux_system_roles.timesync
+  vars:
+    timesync_ntp_servers:
+      - hostname:  "{{ ad_integration_timesync_source }}"
+        iburst: yes
+  when: ad_integration_manage_timesync | bool
+
+- name: Manage crypto policies
+  include_role:
+   name: fedora.linux_system_roles.crypto_policies
+  vars:
+    crypto_policies_policy: "DEFAULT:AD-SUPPORT"
+  when:
+    - ad_integration_manage_crypto_policies | bool
+    # Fedora and RHEL8+
+    - (ansible_distribution == "Fedora" or (ansible_distribution in ['CentOS', 'RedHat'] and ansible_distribution_version is version('8', '>=')))
```

Comment:
this can be wrapped like this:
```yaml
    - (ansible_distribution == "Fedora" or
      (ansible_distribution in ['CentOS', 'RedHat'] and
       ansible_distribution_version is version('8', '>=')))
```

Reply:
Fixed.